src/main/webapp/scripts/map.js

```diff
@@ -94,21 +94,29 @@ function initMap() {
     zoom: 15
   });
   
-  /* Get all stored markers */
+  /* Get all stored recommendation lat, lng and id */
   fetchAndPlaceMarkers(map);
 
   map.addListener("click", e => {
     placeMarkerAndPanTo(e.latLng, map);
   });
-  
-  /* Add Hard-Coded Markers */
-  addHardCodedMarkers(map);
+
+  /* Filter out the recommendations by category if a category parameter is passed */
+  /* this is the case when a category button is clicked */
+  if (category) {
+    addMarker(map, recommendations.filter(recommendation => recommendation.category === category));
+  }
+  /* Add all recommendations if there is no parameter given (when loading the page) */
+  else {
+    addMarker(map, recommendations);
+  }
 }
 
-function addHardCodedMarkers(map) {
-  /* Iterate through stored recommendations to get details */
-  for (i = 0; i < recommendations.length; i++) {
-    const recommendation = recommendations[i];
+/* Add Hard-Coded Markers to map */
+function addMarker(map, recommendationsToAdd) {
+  /* Iterate through stored recommendationsToAdd to get details */
+  for (i = 0; i < recommendationsToAdd.length; i++) {
+    const recommendation = recommendationsToAdd[i];
```

Comment:
ditto, this is already pushed with https://github.com/googleinterns/step241-2020/commit/c1625c8ae4bfb10dae2d93ed3e01ce201445a21c

Reply:
Yes, not sure why it's included